docs/content/sips/013-cloud-plugin.md

```diff
@@ -0,0 +1,155 @@
+
+Owners: matt.fisher@fermyon.com
+
+Created: May 10, 2023
+
+## Background
+
+The intended outcome of this SIP is to provide a set of recommendations to move
+the core logic of `spin login` and `spin deploy` into a separate `spin cloud`
+plugin, while continuing to provide a great out-of-the-box experience for those
+wanting to deploy their applications to the Fermyon Cloud.
+
+## Proposal
+
+This document proposes to move two commands from the Spin command-line interface
+to a separate plugin:
+
+1. `spin login`
+2. `spin deploy`
+
+These commands will be moved to a separate `spin cloud` plugin. This document
+will also recommend possible approaches to ensure these commands remain easily
+accessible to new and existing users of the Spin command-line interface.
+
+This enables the Spin team to release the core Spin runtime and the Spin
+command-line interface as stable, while also enabling the functionality of the
+Fermyon Cloud to change and iterate over time.
+
+## Rationale
+
+Several commands were introduced to the Spin CLI, making it very simple for
+someone to deploy their application to the Fermyon Cloud:
+
+- `spin login`
+- `spin deploy`
+
+These commands are orthogonal to the concerns of Spin's core runtime. These
+commands involve the packaging and distribution of a Spin application to the
+Fermyon Cloud. They are considered "adjacent" to the Spin user experience as
+they do not assist the developer with writing their application, nor do they
+relate to Spin's core runtime; they provide a simple experience to ship their
+application to the Fermyon Cloud.
+
+`spin login` and `spin deploy` communicate with the Fermyon Cloud API. As new
+features are introduced to the Fermyon Cloud, the API may change and evolve over
+time.
+
+Building a simple, delightful on-ramp experience for the Fermyon Cloud remains
+top priority. It is especially important that we continue to provide a very
+simple on-ramp experience so users can readily deploy their Spin applications to
+a production-grade system.
+
+Spin’s existing plug-in system allows the Spin community to add and extend the
+functionality of Spin without introducing changes to the Spin command-line
+interface. Plug-ins are described in more detail in [SIP 006 - Spin
+Plugins](./006-spin-plugins.md). This allows us to ship `spin login` and `spin
+deploy` as separate, discrete functionality while ensuring users can still
+access its functionality through familiar tooling.
+
+## Specification
+
+The proposal herein suggests that we re-release the core logic of `spin login`
+and `spin deploy` under a separate `spin cloud` plugin. Spin will alias the
+existing `spin login` and `spin deploy` commands to their respective `spin
+cloud` counterparts to retain backwards compatibility.
+
+When a user executes `spin login` or `spin deploy` and the cloud plugin is not
+installed on their system, Spin will inform the user that the `spin cloud`
+plugin has not been installed.
+
+```
```

Comment:
@bacongobbler the current implementation takes this further. If the cloud plugin is not installed, it starts installing it. This is the current output:
```
The `cloud` plugin is required. Installing now.
Are you sure you want to install plugin 'cloud' with license Apache-2.0 from file:///Users/kagold/Programs/cloud-plugin/plugin/cloud.tar.gz? yes
Plugin 'cloud' was installed successfully!
```

We could even add the `--yes` flag to avoid prompting the user about the license since it is a Fermyon plugin. Thoughts?

Reply:
Ii will update the SIP to match. Thanks for catching this.

the “yes” flag is an interesting one that spans beyond the scope of this PR. Each “default” plugin could follow the same behaviour. Also by having “yes” as a flag, I don’t know how that would interact with the plugin when it is installed. Is it passed to the plug-in? Is it stripped from the CLI options before invoking the plug-in?

Reply:
@bacongobbler thanks! by the `--yes` flag I mean the `spin plugins install cloud --yes` which will install the plugin without prompting about the license. I updated the implementation to use it by setting `yes_to_all: true`

Reply:
Ah yes that would be fine!